tap_salesforce/salesforce/__init__.py

```diff
@@ -276,23 +276,23 @@ def login(self):
 
         LOGGER.info("Attempting login via OAuth2")
 
-        resp = self.session.post(
-            login_url, data=login_body, headers={
-                "Content-Type": "application/x-www-form-urlencoded"})
-
         try:
+            resp = self.session.post(
+                login_url, data=login_body, headers={"Content-Type": "application/x-www-form-urlencoded"})
+
             resp.raise_for_status()
-        except Exception as e:
-            raise Exception(str(e) + ", Response from Salesforce: {}".format(resp.text)) from e
 
-        LOGGER.info("OAuth2 login successful")
+            LOGGER.info("OAuth2 login successful")
 
-        auth = resp.json()
+            auth = resp.json()
 
-        self.access_token = auth['access_token']
-        self.instance_url = auth['instance_url']
-        self.login_timer = threading.Timer(REFRESH_TOKEN_EXPIRATION_PERIOD, self.login)
-        self.login_timer.start()
+            self.access_token = auth['access_token']
+            self.instance_url = auth['instance_url']
+        except Exception as e:
+            raise Exception(str(e) + ", Response from Salesforce: {}".format(resp.text)) from e
```

Comment:
`resp.text` could possibly throw if session.post throws to get us into this. It's very useful information if it exists, maybe move it into an "if exists" check and add it on if so?